src/main/java/co/rsk/federate/signing/hsm/advanceBlockchain/HsmBookkeepingClientImpl.java

```diff
@@ -28,12 +28,17 @@ public class HsmBookkeepingClientImpl implements HSMBookkeepingClient {
     private final HSMClientProtocol hsmClientProtocol;
     private final int version;
 
-    public HsmBookkeepingClientImpl(HSMClientProtocol hsmClientProtocol, int version) {
+    public HsmBookkeepingClientImpl(HSMClientProtocol hsmClientProtocol) throws HSMClientException {
         this.hsmClientProtocol = hsmClientProtocol;
-        this.version = version;
+        this.version = hsmClientProtocol.getVersion();
```

Comment:
I wouldn't keep this variable here, I think can be error prone. We should use getVersion() method when needed

Reply:
The getVersion() Calls The Device Everytime That Method Is Called.. This Solves The Issue We Are Having With Calling The HSM Device Multiple Times
Another Alternative Is To Set The Version The First Time We Called getVersion From The Protocol Class